applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -37,13 +37,28 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
 #define CONFIG_USB_HID_PROTOCOL_CODE -1
 #endif
 
+#define USB_HID_SELECTIVE_SUB	IS_ENABLED(CONFIG_DESKTOP_USB_SELECTIVE_HID_SUBSCRIPTION)
+
+#if USB_HID_SELECTIVE_SUB
+  #include "usb_state_def.h"
+#else
+  static const uint16_t usb_hid_selected_report_bitmask[0];
+  #if (CONFIG_DESKTOP_HID_STATE_ENABLE) && (CONFIG_USB_HID_DEVICE_COUNT > 1)
+    #error USB selective HID subscription must be enabled.
+  #endif
+#endif /* USB_HID_SELECTIVE_SUB */
+
+#if DT_PROP(DT_NODELABEL(usbd), num_in_endpoints) < (CONFIG_USB_HID_DEVICE_COUNT + 1)
+  #error Too few USB IN Endpoints enabled. Modify dts.overlay file.
+#endif
 
 struct usb_hid_device {
 	const struct device *dev;
 	uint8_t hid_protocol;
 	uint8_t sent_report_id;
 	bool report_enabled[REPORT_ID_COUNT];
 	bool enabled;
+	uint16_t report_bitmask;
```

Comment:
Maybe we could stick to using `uint16_t` here? That could let us save some memory (important on dongles). We still have a few "free" bits  that could be used for new HID reports.

Reply:
In other places we use 32-bit bitmasks so there will be no problems in case more reports are defined in the report map (which is expected as clients add their own reports).
As for the saving you talk about 2 bytes that in practice may mean there will be no saving at all depending on report count. You also talk about 2 byte saving in system that is known to leak hundreds of them between releases.
I prefer consistency and space for additional reports.

Reply:
use shorter name: `report_bm`

Reply:
done